api/src/test/java/uk/gov/dhsc/htbhf/claimant/reporting/payload/ReportPropertiesFactoryTest.java

```diff
@@ -0,0 +1,198 @@
+                entry("cd10", postcodeData.getCodes().getCcg())); // Clinical Commissioning Group code
+        long expectedClaimantAge = Period.between(claim.getClaimant().getDateOfBirth(), timestamp.toLocalDate()).getYears();
+        assertThat(reportProperties.get("cm7")).isEqualTo(String.valueOf(expectedClaimantAge));
+        assertThat(reportProperties).doesNotContainKeys("cm4", "cm5", "cm6", "cm8"); // payment-only custom metrics
+        verify(claimantCategoryCalculator).determineClaimantCategory(claim.getClaimant(), datesOfBirthOfChildren, timestamp.toLocalDate());
+    }
+
+    @Test
+    void shouldImposeMaximumValueForQueueTime() {
+        ReportClaimMessageContext context = aReportClaimMessageContext(LocalDateTime.now().minusHours(5), emptyList(), null);
+        given(claimantCategoryCalculator.determineClaimantCategory(any(), any(), any())).willReturn(CLAIMANT_CATEGORY);
+
+        Map<String, String> reportProperties = reportPropertiesFactory.createReportPropertiesForClaimEvent(context);
+
+        // queue times longer than 4 hours may cause the event to not be registered:
+        // https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#qt
+        Long queueTime = Long.parseLong(reportProperties.get("qt"));
+        long maxQueueTime = 1000 * 60 * 60 * 4;
+        assertThat(queueTime).isLessThanOrEqualTo(maxQueueTime);
+    }
+
+    @Test
+    void shouldIncludeMetricsForZeroChildren() {
+        ReportClaimMessageContext context = aReportClaimMessageContext(LocalDateTime.now(), emptyList(), null);
+        given(claimantCategoryCalculator.determineClaimantCategory(any(), any(), any())).willReturn(CLAIMANT_CATEGORY);
+
+        Map<String, String> reportProperties = reportPropertiesFactory.createReportPropertiesForClaimEvent(context);
+
+        assertThat(reportProperties).contains(
+                entry("cm1", "0"),
+                entry("cm2", "0")
+        );
+    }
+
+    @ParameterizedTest(name = "{1} children under 1 and {2} between 1 and 4 given: {0}")
+    @MethodSource("childrensDatesOfBirth")
+    void shouldIncludeMetricsForNumberOfChildren(List<LocalDate> datesOfBirth, long childrenUnderOne, long childrenOneToFour) {
+        ReportClaimMessageContext context = aReportClaimMessageContext(LocalDateTime.now(), datesOfBirth, null);
+        given(claimantCategoryCalculator.determineClaimantCategory(any(), any(), any())).willReturn(CLAIMANT_CATEGORY);
+
+        Map<String, String> reportProperties = reportPropertiesFactory.createReportPropertiesForClaimEvent(context);
+
+        assertThat(reportProperties).contains(
+                entry("cm1", String.valueOf(childrenUnderOne)),
+                entry("cm2", String.valueOf(childrenOneToFour))
+        );
+    }
+
+    static Stream<Arguments> childrensDatesOfBirth() {
+        return Stream.of(
+                Arguments.of(List.of(LocalDate.now().minusDays(1), LocalDate.now().minusMonths(11)), 2, 0),
```

Comment:
Could we use constants for these dates of birth so that its easier to see the specific meaning of each date as we've done in other tests?

Reply:
Done.